constantine/commitments/ipa/transcript_gen.nim

```diff
@@ -0,0 +1,79 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+# ############################################################
+#
+#         sha256 Generator for Challenge Scalars
+#
+# ############################################################
+
+import
+    ./helper_types,
+    ../../platforms/[primitives,abstractions],
+    ../../serialization/endians,
+    ../../math/config/[type_ff, curves],
+    ../../math/[extension_fields, arithmetic],
+    ../../math/elliptic/ec_twistededwards_projective,
+    ../../../constantine/hashes,
+    ../../../constantine/serialization/[codecs_banderwagon,codecs_status_codes]
+
+
```

Comment:
I think we can create a data structure

```Nim
type IpaTranscript[H: CryptoHash, N: static int] = object
  ctx: H
  label: array[N, byte]
```

And recreate the init/update/finish of the `CryptoHash` concept with just prepending `update(hash)` in front.

Reply:
Added, although I think I can use it in a more refined way.